File: algorithm/Programmers/JAVA/LV4/thief.java
package Programmers.JAVA.LV4;


public class thief {

    public static void main(String[] args){
        int[] money = {91,90,5,7,5,7};

        solution(money);
    }
    
    public static int solution(int[] money) {
        int length = money.length;
        int[] dp =new int[length-1];
        int[] dp2= new int[length];
        
        dp[0]=money[0];
        dp[1]=money[0];
        dp2[0]=0;
        dp2[1]=money[1];
        for(int i=2;i<length-1;i++){
            dp[i]=Math.max(dp[i-2]+money[i],dp[i-1]);
        }
        for(int i=2;i<length;i++){
            dp2[i]=Math.max(dp2[i-2]+money[i],dp2[i-1]);
        }
        
        return Math.max(dp[length-2],dp2[length-1]);
    }
}

File: algorithm/Programmers/Kotlin/LV4/thief.kt
fun main() {
    var money = arrayOf(91, 90, 5, 7, 5, 7)
    println(solution(money))
}

fun solution(money: Array<Int>) : Int{

    var answer = 0
    var length = money.size
    var dp = IntArray(length-1) { 0 }
    var dp2 = IntArray(length) { 0 }

    dp[0] = money[0]
    dp[1] = money[0]
    dp2[0] = 0
    dp2[1] = money[1]

    for(i in 2 until length-1){
        dp[i] = Math.max(dp[i-2] + money[i], dp[i-1])
    }
    for(i in 2 until length){
        dp2[i] = Math.max(dp2[i-2] + money[i], dp2[i-1])
    }

    return Math.max(dp[length-2], dp2[length-1])

}